### PR TITLE
Add RuntimeDefault to ctrcfg

### PR DESCRIPTION
**Pull request by haircommander**
**- What I did**
Update the containerruntimeconfig with `DefaultRuntime` which allows users to specify a different default runtime. At present, the only fully supported options are `"runc"` and `""`. If the cluster has the `TechPreviewNoUpgrade` feature set enabled, then they gain access to `crun` as well.

**- How to verify it**
Launch a cluster with this PR
run `oc describe crd/containerruntimeconfigs.machineconfiguration.openshift.io`
see the presence of `defaultRuntime` field
**- Description for the changelog**

Add support for DefaultRuntime in ContainerRuntimeConfig CRD, allowing users to specify a different default OCI runtime for CRI-O to run containers with. At present, the only fully supported options are `"runc"` and `""`. If the cluster has the `TechPreviewNoUpgrade` feature set enabled, then they gain access to `crun` as well.

### CODE REVIEW

**Comment by haircommander** [comment]
@QiWang19 @rphillips @mrunalp PTAL

**Comment by haircommander** [comment]
tested with cluster bot:
`launch openshift/machine-config-operator#3295`
```
oc describe crd/containerruntimeconfigs.machineconfiguration.openshift.io
...
                  Default Runtime:                                                                                                                                                                                                            
                    Description:  defaultRuntime is the name of the OCI runtime to be used as the default.                                                                                                                                    
                    Pattern:      ^$|^runc$
...
```
vs 

`launch openshift/machine-config-operator#3295 techpreview`
```
oc describe crd/containerruntimeconfigs.machineconfiguration.openshift.io
...
                  Default Runtime:                                                                                                                                                                                                            
                    Description:  defaultRuntime is the name of the OCI runtime to be used as the default.                                                                                                                                    
                    Pattern:      ^$|$(runc|crun)$
...
```

**Comment by haircommander** [comment]
@sinnykumari @yuqi-zhang @cheesesashimi @jkyros PTAL

**Review by QiWang19 — COMMENTED** [review]
LGTM
One nit, do we need to specify the supported runtime options in the `description`?

**Comment by haircommander** [comment]
> LGTM One nit, do we need to specify the supported runtime options in the `description`?

I don't think so, because we validate the allowed values with the API. `oc describe crd/....` (I forget the precise crd name) shows the regex that's validated against

File: install/0000_80_machine-config-operator_02_containerruntimeconfig.crd.yaml [hunk 1]
```diff
@@ -0,0 +1,175 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: containerruntimeconfigs.machineconfiguration.openshift.io
```
Comment (yuqi-zhang) [inline install/0000_80_machine-config-operator_02_containerruntimeconfig.crd.yaml:4]: I'm a bit confused, isn't this file mostly just a copy of 01_containerruntineconfig.crd.yaml? Why is this split out into a second file?

Comment (haircommander) [reply install/0000_80_machine-config-operator_02_containerruntimeconfig.crd.yaml:4]: the operative lines are `release.openshift.io/feature-gate: TechPreviewNoUpgrade` on line 11 and `pattern: ^$|^(runc|crun)$` on line 89. Basically, this crd enables `crun` as an option, but only when tech preview is enabled on the node

Comment (yuqi-zhang) [reply install/0000_80_machine-config-operator_02_containerruntimeconfig.crd.yaml:4]: Right, I think I might be confused here. I was under the impression the `02` crd will simply overwrite `01` regardless of whether the feature gate itself is active or not. It looks like from

https://github.com/openshift/machine-config-operator/pull/3295#issuecomment-1222772121

That is not the case and it actually deploys a different CRD based on the conditional.

Comment (haircommander) [reply install/0000_80_machine-config-operator_02_containerruntimeconfig.crd.yaml:4]: CVO magic :smiley:

**Pull request merged**